src/incubator/WheelPicker/index.tsx

```diff
@@ -39,47 +46,104 @@ export interface WheelPickerProps {
   /**
    * Event, on active row change
    */
-  onChange: (index: number, item?: ItemProps) => void;
+  onValueChange?: (item: string | number, index: number) => void;
+  /**
+   * Container's ViewStyle, height is computed according to itemHeight * numberOfVisibleRows
+   */
+  style?: Omit<ViewStyle, 'height'>;
+  /**
+   * Support passing items as children props
+   */
+  children?: JSX.Element | JSX.Element[];
+  /**
+   * WheelPicker initial value, can be ItemProps.value, number as index
+   */
+  selectedValue: ItemProps | number | string;
 }
 
-const WheelPicker = ({items, itemHeight = 48, activeTextColor, inactiveTextColor, textStyle, onChange: onChangeEvent}: WheelPickerProps) => {
-  const height = itemHeight * 5;
+const WheelPicker = ({
+  items: propItems,
+  itemHeight = 44,
+  numberOfVisibleRows = 5,
+  activeTextColor,
+  inactiveTextColor,
+  textStyle,
+  onValueChange,
+  style,
+  children,
+  selectedValue
+}: WheelPickerProps) => {
   const scrollView = useRef<Animated.ScrollView>();
   const [offset] = useValues([0], []);
+  let scrollOffset = useRef<number>(0).current;
   const onScroll = onScrollEvent({y: offset});
-  
-  const listSize = items?.length || 0;
-  const middleIndex = useMiddleIndex({itemHeight, listSize});
+
+  const {height, items, shouldControlComponent, index: currentIndex, getRowItemAtOffset} = usePresenter({
```

Comment:
Is `currentIndex` actually represents the initial index?

Reply:
Not the initial, the current index (`onChange` updates the presenter with the current offset, and the presenter calculates it based on the `selectedValue` )